pkg/uhttp/gocache.go

```diff
@@ -0,0 +1,170 @@
+package uhttp
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"io"
+	"net/http"
+	"net/http/httputil"
+	"time"
+
+	bigcache "github.com/allegro/bigcache/v3"
+	"github.com/grpc-ecosystem/go-grpc-middleware/logging/zap/ctxzap"
+	"go.uber.org/zap"
+)
+
+type GoCache struct {
+	ttl         time.Duration
+	rootLibrary *bigcache.BigCache
+}
+
+func NewGoCache(ctx context.Context, ttl int32, cacheMaxSize int, isLogLevel bool) (GoCache, error) {
+	l := ctxzap.Extract(ctx)
+	config := bigcache.Config{
```

Comment:
I think it's less code to initialize config to `bigcache.DefaultConfig(time.Duration(ttl) * time.Second)`, then override the config fields we want to override.

Reply:
Fixed